bothan-binance/src/store.rs

```diff
@@ -0,0 +1,137 @@
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+
+use tokio::sync::mpsc::{channel, Sender};
+use tokio::sync::Mutex;
+use tokio::sync::RwLock;
+
+use asset_store::start_asset_store;
+use bothan_core::store::{AssetStatus, AssetStore, Error as StoreError};
+use bothan_core::types::AssetInfo;
+
+use crate::api::websocket::{BinanceWebSocketConnection, BinanceWebSocketConnector};
+
+mod asset_store;
+pub mod builder;
+mod types;
+
+/// A store that fetches and stores the asset information from Binance's API.
+pub struct BinanceStore {
+    connector: Arc<BinanceWebSocketConnector>,
+    connection: Arc<Mutex<BinanceWebSocketConnection>>,
+    data_store: Arc<RwLock<HashMap<String, AssetInfo>>>,
+    query_ids: Arc<RwLock<HashSet<String>>>,
+    internal_ch_size: usize,
+    subscribe_ch_tx: Option<Sender<Vec<String>>>,
```

Comment:
I really don't like these channels being `Option`. Can we do this in `new` (require new to be async) so this is not option? or otherwise create another type after `start` call.